server/routes/offer.routes.js

```diff
@@ -84,7 +84,13 @@ router.put('/accept', authenticate, schemaValidation(acceptOfferSchema), OfferCo
  *      '500':
  *       description: Internal server error
  */
-router.put('/cancel', authenticate, schemaValidation(respondOfferSchema), OfferController.cancel);
+router.put(
+  '/cancel',
+  authenticate,
+  isAdmin,
+  schemaValidation(respondOfferSchema),
```

Comment:
The name here is a little confusing, in we change it into something more descriptive. I am afraid that in 2 weeks time, I would have forgotten the purpose of the schema without checking the code 😅 

```suggestion
  schemaValidation(validateOfferIdSchema),
```